pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -180,16 +180,15 @@ func convertRbacRulesToFilterConfig(service string, roles []model.Config, bindin
 		Policies: map[string]*policyproto.Policy{},
 	}
 
+	permissiveRbac := &policyproto.RBAC{
+		Action:   policyproto.RBAC_ALLOW,
+		Policies: map[string]*policyproto.Policy{},
+	}
+
 	for _, role := range roles {
-		log.Debugf("checking role %v for service %v", role.Name, service)
-		principals := convertToPrincipals(roleToBinding[role.Name])
-		if len(principals) == 0 {
-			// Skip to next role if found no bindings for current role. This means nobody could
-			// access the current role, so we don't need to check the remaining rules anymore.
-			log.Debugf("role %v skipped for no bindings found", role.Name)
-			continue
-		}
+		enforcedPrincipals, permissivePrincipals := convertToPrincipals(role.Spec.(*rbacproto.ServiceRole).Mode, roleToBinding[role.Name])
 
+		log.Debugf("checking role %v for service %v", role.Name, service)
```

Comment:
[optional] Add
```
if checklen(enforcedPrincipals) ==0 && len(permissivePrincipals) == 0 {
  continue
}
```
for early exit if the lists are empty.

Reply:
done